### PR TITLE
show the entire stack when there's an error

### PR DESCRIPTION
**Pull request by devTristan**
related to #68 

I just made it print the entire stack trace.

Sorry about the removal of the trailing spaces, I can't figure out how to get my editor to not do that when I save.

### CODE REVIEW

File: lib/results.js [hunk 1]
```diff
@@ -93,7 +93,7 @@ Results.prototype._watch = function (t) {
     t.once('prerun', function () {
         write('# ' + t.name + '\n');
     });
-    
+
```
Comment (ljharb) [inline lib/results.js]: please revert all these whitespace changes; this is part of the project's style

File: lib/results.js [hunk 2]
```diff
@@ -127,23 +127,28 @@ Results.prototype.close = function () {
 function encodeResult (res, count) {
     var output = '';
     output += (res.ok ? 'ok ' : 'not ok ') + count;
-    output += res.name ? ' ' + res.name.toString().replace(/\s+/g, ' ') : '';
-    
+
+    if (typeof res.name === 'object' && typeof res.name.stack === 'string') {
+      output += ' ' + res.name.stack;
+    } else if (res.name) {
+      output += ' ' + res.name.toString().replace(/\s+/g, ' ');
```
Comment (ljharb) [inline lib/results.js]: `String(foo)` is more reliable than `foo.toString()`, please use that instead

**Comment by devTristan** [comment]
:+1: fixed the whitespace and switched to `String(res.name)`

**Comment by devTristan** [comment]
@ljharb [ gentle ping ]

**Comment by ljharb** [comment]
@devTristan sorry about the delay! Would you mind rebasing on latest master? I think this will also need tests ensuring that the stack trace behavior appears when expected, and not when not.

**Comment by devTristan** [comment]
@ljharb alright, I've done the rebase and added a test. Unfortunately this change breaks other tests which do not expect to see a stack trace when an assertion fails. I don't have time to fix them right now, but hopefully I can get to it soon.

**Comment by OliverJAsh** [comment]
This can be closed now with #330 merged?

**Pull request closed without merging**